modules/data.land/R/soil_utils.R

```diff
@@ -264,7 +264,16 @@ sclass <- function(sandfrac,clayfrac){
   #----- Define the percentage of sand, clay, and silt. ----------------------------------#
   sand <- 100. * sandfrac
   clay <- 100. * clayfrac
-  silt <- 100. - sand - clay
+  # Prevent silt from being negative due to numerical precision issue
+  silt <- c()
+  for (i in seq_along(sand)) {
+    if (sand[i] + clay[i] >= 100.) {
```

Comment:
if sand + clay > 100 shouldn't it be renormalized?

```
if sand + clay > 100{
  sand <- sand/(sand+clay) * 100
  clay <- clay/(sand+clay) * 100
}
```

and is this an error in SoilGrids that warrants contacting the creators?

Reply:
Actually, now I see this normalizing is done in the rescale_sum_to_one function; would it make sense to add this silt correction that function?

Reply:
The normalizing in rescale_sum_to_one mainly solved the data issue (the sum of means doesn't equal 1) of SoilGrids. But the codes in soil_utils.R are only for the R numeric rounding issue as the texture percentages used here are from Dirichlet distribution sampling and their sums strictly equal one. So I guess non-normalization here seems fine.

> Actually, now I see this normalizing is done in the rescale_sum_to_one function; would it make sense to add this silt correction that function?